lms/djangoapps/instructor/settings/common.py

```diff
@@ -10,35 +10,85 @@ def plugin_settings(settings):
     settings.ANALYTICS_DASHBOARD_URL = ""
     settings.ANALYTICS_DASHBOARD_NAME = _('Your Platform Insights')
     settings.FEATURES.update({
-        # Enable display of enrollment counts in instructor dash, analytics section
+        # .. toggle_name: FEATURES['DISPLAY_ANALYTICS_ENROLLMENTS']
+        # .. toggle_implementation: DjangoSetting
+        # .. toggle_default: True
+        # .. toggle_description: Enable display of enrollment counts in instructor dashboard and
+        #   analytics section.
+        # .. toggle_use_cases: opt_out
+        # .. toggle_creation_date: 2014-11-12
+        # .. toggle_tickets: https://github.com/edx/edx-platform/pull/5838
         'DISPLAY_ANALYTICS_ENROLLMENTS': True,
 
-        # Display the 'Analytics' tab in the instructor dashboard for CCX courses.
-        # Note: This has no effect unless ANALYTICS_DASHBOARD_URL is already set,
-        #       because without that setting, the tab does not show up for any courses.
+        # .. toggle_name: FEATURES['ENABLE_CCX_ANALYTICS_DASHBOARD_URL']
+        # .. toggle_implementation: DjangoSetting
+        # .. toggle_default: False
+        # .. toggle_description: Display the 'Analytics' tab in the instructor dashboard for CCX courses.
+        #   Note: This has no effect unless ANALYTICS_DASHBOARD_URL is already set, because without that
+        #   setting, the tab does not show up for any courses.
+        # .. toggle_use_cases: opt_in
+        # .. toggle_creation_date: 2016-10-07
+        # .. toggle_tickets: https://github.com/edx/edx-platform/pull/13196
         'ENABLE_CCX_ANALYTICS_DASHBOARD_URL': False,
 
-        # Disable instructor dash buttons for downloading course data
-        # when enrollment exceeds this number
+        # .. setting_name: FEATURES['MAX_ENROLLMENT_INSTR_BUTTONS']
+        # .. setting_default: 200
+        # .. setting_description: Disable instructor dashboard buttons for downloading course data
+        #   when enrollment exceeds this number. The number indicates the maximum allowed enrollments
+        #   for the course to be considered "small". Courses exceeding the upper limit of "small"
+        #   courses will have disabled buttons at the instructor dashboard.
         'MAX_ENROLLMENT_INSTR_BUTTONS': 200,
 
-        # Grade calculation started from the instructor dashboard will write grades
-        # CSV files to the configured storage backend and give links for downloads.
+        # .. toggle_name: FEATURES['ENABLE_GRADE_DOWNLOADS']
+        # .. toggle_implementation: DjangoSetting
+        # .. toggle_default: False
+        # .. toggle_description: Enable grade CSV downloads from the instructor dashboard. Grade
+        #   calculation started from the instructor dashboard will write grades CSV files to the
+        #   configured storage backend and give links for downloads.
+        # .. toggle_use_cases: opt_in
```

Comment:
In general, we are going to err on marking toggles `temporary` if there is any doubt, to flag it for investigation.

If I propose marking this as `temporary` with a new `toggle_warning` with something like: "This may have been a temporary rollout toggle. A future DEPR might investigate removing and enabling for all.", do you information from your research that shows that this must be kept a toggle? Note: if we do decide to do this, we should just add the target removal date as creation date + 3 months.

On the other hand, if this is not needed or wanted for some installations, maybe you could enhance the docs to note when you would or would not want this?

Reply:
FYI: Marking this optional work as resolved so we can more quickly land this.